samples/siwx91x_ota/src/main.c

```diff
@@ -0,0 +1,751 @@
+		net_addr_ntop(AF_INET6, &net_sin6(entry)->sin6_addr, addr, sizeof(addr));
+	} else {
+		strcpy(addr, "<unknown format>");
+	}
+	printf("%s DNS Address: %s\n", op, addr);
+}
+
+static void ota_handle_wifi_scan_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_scan_result *entry = (const struct wifi_scan_result *)cb->info;
+	uint8_t mac_string_buf[sizeof("xx:xx:xx:xx:xx:xx")];
+	uint8_t ssid_print[WIFI_SSID_MAX_LEN + 1];
+	struct app_ctx *ctx = CONTAINER_OF(cb, struct app_ctx, wlan_mgmt_cb);
+
+	if (strncmp(entry->ssid, CONFIG_OTA_WIFI_SSID, entry->ssid_length) == 0) {
+		ctx->security_type = entry->security;
+	}
+
+	strncpy(ssid_print, entry->ssid, sizeof(ssid_print) - 1);
+	ssid_print[sizeof(ssid_print) - 1] = '\0';
+
+	if (entry->mac_length) {
+		snprintf(mac_string_buf, sizeof(mac_string_buf), "%02x:%02x:%02x:%02x:%02x:%02x",
+			 entry->mac[0], entry->mac[1], entry->mac[2], entry->mac[3], entry->mac[4],
+			 entry->mac[5]);
+	} else {
+		mac_string_buf[0] = '\0';
+	}
+
+	printf("%-32s %-5u | %-4u (%-6s) | %-4d | %-15s | %-17s | %-8s\n", ssid_print,
+	       entry->ssid_length, entry->channel, wifi_band_txt(entry->band), entry->rssi,
+	       wifi_security_txt(entry->security), mac_string_buf, wifi_mfp_txt(entry->mfp));
+}
+
+static void ota_handle_wifi_scan_done(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+	struct app_ctx *ctx = CONTAINER_OF(cb, struct app_ctx, wlan_mgmt_cb);
+
+	ctx->scan_result_count = 0;
+	if (status->status) {
+		printf("Scan request failed (%d)\n", status->status);
+	}
+}
+
+static void ota_handle_wifi_connect_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+	struct app_ctx *ctx = CONTAINER_OF(cb, struct app_ctx, wlan_mgmt_cb);
+
+	if (status->status) {
+		printf("Connection request failed\n");
+	} else {
+		printf("Connected to Wi-Fi\n");
+	}
+	ctx->disonnected = false;
+}
+
+static void ota_handle_wifi_disconnect_result(struct net_mgmt_event_callback *cb)
+{
+	const struct wifi_status *status = (const struct wifi_status *)cb->info;
+	struct app_ctx *ctx = CONTAINER_OF(cb, struct app_ctx, wlan_mgmt_cb);
+
+	if (status->status) {
+		printf("Disconnection request failed (%d)\n", status->status);
+		ctx->disonnected = false;
+	} else {
+		printf("Disconnection reason: %d\n", status->disconn_reason);
+		ctx->disonnected = true;
+	}
+}
+
+static void ota_wifi_mgmt_event_handler(struct net_mgmt_event_callback *cb, uint64_t mgmt_event,
+					struct net_if *iface)
+{
+
+	switch (mgmt_event) {
+	case NET_EVENT_WIFI_SCAN_RESULT:
+		ota_handle_wifi_scan_result(cb);
+		break;
+	case NET_EVENT_WIFI_SCAN_DONE:
+		ota_handle_wifi_scan_done(cb);
+		break;
+	case NET_EVENT_WIFI_CONNECT_RESULT:
+		ota_handle_wifi_connect_result(cb);
+		break;
+	case NET_EVENT_WIFI_DISCONNECT_RESULT:
+		ota_handle_wifi_disconnect_result(cb);
+		break;
+	default:
+		break;
+	}
+}
```

Comment:
These indirections have no benefit. Just register the `ota_handle_wifi_*()` function to the right event and you will be able to drop this function.